code/lib/create-storybook/src/generators/REACT_SCRIPTS/index.ts

```diff
@@ -2,69 +2,66 @@ import { existsSync } from 'node:fs';
-        if (Array.isArray(p.appSrcs)) {
-          p.appSrcs.push('${monorepoRootPath}');
-              }
-            });
-          return config;
-          }
-    %%`,
-      }
-    : {};
+export default defineGeneratorModule({
+  metadata: {
+    projectType: ProjectType.REACT_SCRIPTS,
+    renderer: SupportedRenderer.REACT,
+    builderOverride: SupportedBuilder.WEBPACK5,
+  },
+  configure: async (packageManager, context) => {
+    const monorepoRootPath = fileURLToPath(new URL('../../../../../../..', import.meta.url));
+    const extraMain = context.linkable
+      ? {
+          webpackFinal: `%%(config) => {
+        // add monorepo root as a valid directory to import modules from
+        config.resolve.plugins.forEach((p) => {
+          if (Array.isArray(p.appSrcs)) {
+            p.appSrcs.push('${monorepoRootPath}');
+                }
+              });
+            return config;
+            }
+      %%`,
+        }
+      : {};
 
-  const craVersion = (await packageManager.getModulePackageJSON('react-scripts'))?.version ?? null;
+    const craVersion =
+      (await packageManager.getModulePackageJSON('react-scripts'))?.version ?? null;
 
-  if (craVersion === null) {
-    throw new Error(dedent`
-      It looks like you're trying to initialize Storybook in a CRA project that does not have react-scripts installed.
-      Please install it and make sure it's of version 5 or higher, which are the versions supported by Storybook 7.0+.
-    `);
-  }
-
-  if (!craVersion && semver.gte(craVersion, '5.0.0')) {
-    throw new Error(dedent`
-      Storybook 7.0+ doesn't support react-scripts@<5.0.0.
-
-      https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#create-react-app-dropped-cra4-support
-    `);
-  }
+    if (craVersion === null) {
+      throw new Error(dedent`
+        It looks like you're trying to initialize Storybook in a CRA project that does not have react-scripts installed.
+        Please install it and make sure it's of version 5 or higher, which are the versions supported by Storybook 7.0+.
+      `);
+    }
 
-  const extraPackages = [];
-  extraPackages.push('webpack');
-  // TODO: Evaluate if this is correct after removing pnp compatibility code in SB11
-  // Miscellaneous dependency to add to be sure Storybook + CRA is working fine with Yarn PnP mode
-  extraPackages.push('prop-types');
+    if (!craVersion && semver.gte(craVersion, '5.0.0')) {
+      throw new Error(dedent`
+        Storybook 7.0+ doesn't support react-scripts@<5.0.0.
+  
+        https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#create-react-app-dropped-cra4-support
+      `);
+    }
 
-  const extraAddons = [`@storybook/preset-create-react-app`];
+    // TODO: Evaluate if adding prop-types is correct after removing pnp compatibility code in SB11
+    // Miscellaneous dependency to add to be sure Storybook + CRA is working fine with Yarn PnP mode
+    const extraPackages = ['webpack', 'prop-types'];
```

Comment:
Does anyone still use prop-types for new projects? I guess maybe we're just keeping this around for sandboxes?